src/Adapter/MSTestAdapter.PlatformServices/Interfaces/ITestContext.cs

```diff
@@ -36,6 +36,12 @@ public interface ITestContext
     /// <param name="outcome"> The outcome. </param>
     void SetOutcome(UnitTestOutcome outcome);
 
+    /// <summary>
+    /// Sets the exception that occurred during the TestInitialize or TestMethod step (if any).
+    /// </summary>
+    /// <param name="exception">The exception.</param>
+    void SetException(Exception? exception);
```

Comment:
It seems like a breaking change. Not sure if okay.

Reply:
Good point! I couldn't find any ref to it using grepapp and as far as I can see in the codebase this is part of the types that are public for unknown reason as there is no way for user to provide a custom test context.